### ISSUE TITLE
V-JEPA 2.1: Config/checkpoint mismatches prevent loading ViT-L/ViT-B checkpoints

### ISSUE THREAD

**Issue by cdboer** [facebookresearch/vjepa2#149]
Several issues we encountered when loading the released V-JEPA 2.1 distilled checkpoints (`vjepa2_1_vitl_dist_vitG_384.pt`, `vjepa2_1_vitb_dist_vitG_384.pt`) for continued pretraining:

### 1. `pred_depth` mismatch (ViT-L config)

The ViT-L config ([`configs/train_2_1/vitl16/pretrain-256px-16f.yaml`](https://github.com/facebookresearch/vjepa2/blob/204698b45b3712590f06245fbfba32d3be539812/configs/train_2_1/vitl16/pretrain-256px-16f.yaml#L123)) specifies `pred_depth: 24` but the released checkpoint has 12 predictor blocks (as expected per the paper's distillation protocol: "we retrain a new predictor, which has 12 blocks instead of 24 for pretraining"). This causes a silent partial load; 12 blocks are loaded while blocks 12-23 remain randomly initialized, with no error raised (`strict=False`). The mismatch is visible in INFO-level log messages listing missing keys.

The ViT-B config correctly specifies `pred_depth: 12`.

### 2. Missing `vit_base` in embed_dim lookup (see also #147)

[`app/vjepa_2_1/train.py` lines 117-124](https://github.com/facebookresearch/vjepa2/blob/204698b45b3712590f06245fbfba32d3be539812/app/vjepa_2_1/train.py#L117-L124) hardcode `embed_dim_encoder` for `vit_large`, `vit_giant_xformers`, and `vit_gigantic_xformers` only. `vit_base` is not handled despite having a released config and checkpoint, leaving `embed_dim_encoder` undefined.

### 3. `ema_encoder` vs `target_encoder` key

The checkpoints store the Polyak-averaged encoder under the key `ema_encoder`, but `load_checkpoint()` in [`app/vjepa_2_1/utils.py` (line 150)](https://github.com/facebookresearch/vjepa2/blob/204698b45b3712590f06245fbfba32d3be539812/app/vjepa_2_1/utils.py#L150) expects `target_encoder`, causing a `KeyError`.

Per the paper: "a separate EMA of the context encoder is maintained, corresponding to Polyak averaging, which is never used in the loss calculation but serves as the final, high-performing model released for downstream tasks." The `ema_encoder` feels like the intended model for continued pretraining.

### 4. Question regarding continued pretraining from distilled checkpoints

The released ViT-B/L checkpoints were trained via distillation from a frozen ViT-G teacher, but the released code only supports self-distillation (EMA target encoder). The predictor's output projection has dim 1664 (matching ViT-G's representation space), which is incompatible with self-distillation where the predictor must project to the student's own representation space (e.g., 4×1024=4096 for ViT-L).

For continued pretraining without a ViT-G teacher, we found the following approach works:
- Load the `ema_encoder` as both the student encoder and target encoder
- Partially load the predictor (transformer blocks 0-11 match), reinitialize mismatched layers (`predictor_embed`, `predictor_proj`, `predictor_proj_context`)
- Reset epoch counter to 0

Is this the recommended approach, or is there a better way to continue training from these distilled checkpoints? 

## Verification

```python
import torch

for name, path in [
    ("ViT-B", "vjepa2_1_vitb_dist_vitG_384.pt"),
    ("ViT-L", "vjepa2_1_vitl_dist_vitG_384.pt"),
]:
    ckpt = torch.load(path, map_location="cpu", weights_only=False)
    pred_keys = list(ckpt["predictor"].keys())
    blocks = sorted(set(int(k.split(".")[3]) for k in pred_keys if "predictor_blocks." in k))
    print(f"{name}: Predictor blocks in checkpoint: {len(blocks)} (indices {blocks[0]}-{blocks[-1]})")
    print(f"{name}: Top keys: {sorted(ckpt.keys())}")
```

```
ViT-B: Predictor blocks in checkpoint: 12 (indices 0-11)
ViT-B: Top keys: ['batch_size', 'ema_encoder', 'encoder', 'epoch', 'loss', 'lr', 'opt', 'predictor', 'scaler', 'world_size']
ViT-L: Predictor blocks in checkpoint: 12 (indices 0-11)
ViT-L: Top keys: ['batch_size', 'ema_encoder', 'encoder', 'epoch', 'loss', 'lr', 'opt', 'predictor', 'scaler', 'world_size']
